container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java

```diff
@@ -240,6 +263,12 @@ private void setInRotationOnlyIf(boolean inRotation) {
             vipStatus.removeFromRotation(clusterId);
     }
 
+    public int estimateHitsToFetch(int wantedHits, int numPartitions) {
+        return ((hitEstimator == null) || (numPartitions <= 1))
```

Comment:
Same, why not move this into the estimator.

Reply:
Might be quite useful to control this by query? How much you care can depend on use case ...

Reply:
Moved.
Will add query too.

Reply:
Now I have added query side control. dispatch.top-k-probability.
Can '-' be used in query without requiring any escaping ?
I used that to have it identical with services.xml.